tso-server/src/main/java/org/apache/omid/tso/TSOChannelHandler.java

```diff
@@ -59,43 +61,37 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(TSOChannelHandler.class);
 
-    private final ChannelFactory factory;
-
     private final ServerBootstrap bootstrap;
-
-    @VisibleForTesting
-    Channel listeningChannel;
-    @VisibleForTesting
-    ChannelGroup channelGroup;
-
-    private RequestProcessor requestProcessor;
-
-    private TSOServerConfig config;
-
-    private MetricsRegistry metrics;
+    private ChannelGroup channelGroup;
+    private boolean closed = false;
+    private final RequestProcessor requestProcessor;
+    private final TSOServerConfig config;
+    private final MetricsRegistry metrics;
 
     @Inject
     public TSOChannelHandler(TSOServerConfig config, RequestProcessor requestProcessor, MetricsRegistry metrics) {
-
         this.config = config;
         this.metrics = metrics;
         this.requestProcessor = requestProcessor;
         // Setup netty listener
-        this.factory = new NioServerSocketChannelFactory(
+        this.bootstrap = new ServerBootstrap(new NioServerSocketChannelFactory(
```

Comment:
There is quite bit a refactoring here that is not strictly related to the test failure.
Please update the ticket summary and/or description to mention this.